### PR TITLE
Update globals in plugin

### PR DESCRIPTION
**Pull request by tomekzaw**
## Summary

This PR adds the names of missing standard built-in objects (e.g. `BigInt` or `WeakSet`) in Reanimated Babel plugin as well as removes redundant ones (e.g. `LayoutAnimationRepository` or `__ErrorUtils`).

Based on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects.

## Test plan

Verify each name one-by-one.

### CODE REVIEW

File: plugin/src/commonObjects.ts [hunk 1]
```diff
@@ -1,63 +1,141 @@
...
+  // 'Intl.ListFormat',
+  // 'Intl.Locale',
+  // 'Intl.NumberFormat',
+  // 'Intl.PluralRules',
+  // 'Intl.RelativeTimeFormat',
+  // 'Intl.Segmenter',
+
+  // Other stuff
   'null',
-  'UIManager',
```
Comment (tjzel) [inline plugin/src/commonObjects.ts:26]: is this legacy?

File: plugin/src/commonObjects.ts [hunk 2]
```diff
@@ -1,63 +1,141 @@
...
+  'AsyncGenerator',
+  'AsyncFunction',
+
+  // Reflection
+  'Reflect',
+  'Proxy',
+
+  // Internationalization
+  'Intl',
+  // 'Intl.Collator',
```
Comment (tjzel) [inline plugin/src/commonObjects.ts:97]: Why is this commented out?

File: plugin/src/commonObjects.ts [hunk 3]
```diff
@@ -1,63 +1,141 @@
...
   '_scheduleOnJS',
   '_makeShareableClone',
   '_updateDataSynchronously',
-  'eval',
   '_updatePropsPaper',
   '_updatePropsFabric',
   '_removeFromPropsRegistry',
-  'RegExp',
-  'Error',
-  '__ErrorUtils',
```
Comment (tjzel) [inline plugin/src/commonObjects.ts:51]: Are you sure we can get rid of __ErrorUtils here?

File: plugin/src/commonObjects.ts (hunk 1 above)
Comment (tomekzaw) [reply plugin/src/commonObjects.ts:26]: It looks so, this line was introduced in "༼ つ ◕_◕ ༽つ Give Reanimated 2" commit and currently there's no occurrences of `UIManager` in any file with extension `*.ts,*.tsx,*.js`.

File: plugin/src/commonObjects.ts (hunk 2 above)
Comment (tomekzaw) [reply plugin/src/commonObjects.ts:97]: If `Intl.Collator` appears in worklet body, our plugin will capture only `Intl`, thus it doesn't make sense to allowlist `Intl.Collator`.

File: plugin/src/commonObjects.ts (hunk 3 above)
Comment (tomekzaw) [reply plugin/src/commonObjects.ts:51]: Yep, it's only used as `global.__ErrorUtils` so the plugin would capture only `global` but this is already allowlisted so it won't. I've tested it using one demo from WorkletExample.tsx which throws JS Errors on the UI runtime.

**Review by tjzel — APPROVED** [review]
🎬

**Pull request merged**